packages/bundlers/default/src/DefaultBundler.js

```diff
@@ -826,7 +745,34 @@ function createIdealGraph(
               }
 
               assetReference.get(childAsset).push([dependency, bundle]);
-              continue;
+            } else {
+              bundleId = null;
```

Comment:
Without this we hit a nullthrows/ assertion error internally, I assume because of the scope change of `bundleId`.